### ISSUE TITLE
Wrong ETag for stations#index causes cache issue.

### ISSUE THREAD

**Issue by maxcal** [remote-wind/remote-wind#124]
Basically this action uses the stations `updated_at` value as a cache key - Don't know what I was thinking when I originally wrote it. 

Since adding observations does not update the station (nor should it since you want to avoid a slow UPDATE query) the client gets a `last_modified` which usually is much older than the latest observation data.

What I was trying do was avoid the query which fetches stations with one observation since it was very  costly and difficult to optimize since the ActiveRecord `.join` and `.include` methods pull all the associated records out of the DB.
 I managed to get the performance pretty decent so its up to you @karl-petter if you want to skip the e-tags completely and send no-cache headers. The server load / client performance hits are probably negligible. 

```

before_action :no_cache, only: [:index]

# ...
private
def no_cache
  response.headers["Cache-Control"] = "no-cache, no-store, must-revalidate" # HTTP 1.1.
  response.headers["Pragma"] = "no-cache" # HTTP 1.0.
  response.headers["Expires"] = "0" # Proxies.
end
```

Or try fixing it to send a correct e-tag:

```
# wrong! @last_updated = Station.order("updated_at asc").last
@last_updated = Observation.last # right
```

```
  # GET /stations
  # GET /stations.json
  def index
    @title = "Stations"
    @last_updated = Observation.last
    if stale?(@last_updated)
      @stations = all_with_latest_observation
      respond_to do |format|
        format.html
        format.json { render json: @stations }
      end
    end
  end
```

**Comment by maxcal** [comment]
I gave this a bit of thought, "touching" the Station when creating observations might not be so bad anyways:

```RUBY
has_many  :observations,
    inverse_of: :station,
    counter_cache: true,
    after_add: ->(s,o){ s.touch } # touch station so cache key is changed
```

It will let you use caching without a bunch of hassle or setting up custom cache keys.

```
if stale?(@station) # uses @station.updated_at for cache keys and last_modified
  # ...
else
  # ...
end
```

 It can also simplify the logic in a bunch as you can use `updated_at` instead of `observations.last.updated_at` which can cause n+1 issues. 

The extra UPDATE query is probably not that much of an issue.

**Comment by maxcal** [comment]
Fixed by #128
